ModularTegustation/tegu_items/lc13unique_items.dm

```diff
@@ -0,0 +1,499 @@
+		if(!ishuman(M) && bullettype == 5)
+			M.apply_status_effect(/datum/status_effect/qliphothoverload)
+			qdel(src)
+			return
+		if(ishuman(M))
+			switch(bullettype)
+				if(1)
+					M.apply_status_effect(/datum/status_effect/interventionshield)
+				if(2)
+					M.apply_status_effect(/datum/status_effect/interventionshield/white)
+				if(3)
+					M.apply_status_effect(/datum/status_effect/interventionshield/black)
+				if(4)
+					M.apply_status_effect(/datum/status_effect/interventionshield/pale)
+				if(5)
+					M.apply_status_effect(/datum/status_effect/qliphothoverload)
+		qdel(src)
+	..()
+
+/datum/status_effect/interventionshield
+	id = "physical intervention shield"
+	duration = 15 SECONDS
+	status_type = STATUS_EFFECT_REPLACE
+	alert_type = null
+	var/inherentarmorcheck
+	var/statuseffectvisual = icon('ModularTegustation/Teguicons/tegu_effects.dmi', "red_shield")
+	var/shieldhealth = 100
+	var/damagetaken = 0
+	var/respectivedamage = RED_DAMAGE
+	var/faltering = 0
+
+/datum/status_effect/interventionshield/on_apply()
+	. = ..()
+	owner.add_overlay(statuseffectvisual)
+	var/mob/living/carbon/human/L = owner
+	switch(respectivedamage)
+		if(RED_DAMAGE)
+			inherentarmorcheck = L.physiology.red_mod
+			L.physiology.red_mod *= 0.0001
+		if(WHITE_DAMAGE)
+			inherentarmorcheck = L.physiology.white_mod
+			L.physiology.white_mod *= 0.0001
+		if(BLACK_DAMAGE)
+			inherentarmorcheck = L.physiology.black_mod
+			L.physiology.black_mod *= 0.0001
+		if(PALE_DAMAGE)
+			inherentarmorcheck = L.physiology.pale_mod
+			L.physiology.pale_mod *= 0.0001
+	owner.visible_message("<span class='notice'>[owner]s shield activates!</span>")
+	RegisterSignal(owner, COMSIG_MOB_APPLY_DAMGE, .proc/OnApplyDamage) //stolen from caluan
+	RegisterSignal(owner, COMSIG_WORK_STARTED, .proc/Destroy)
+
+/datum/status_effect/interventionshield/proc/OnApplyDamage(datum_source, amount, damagetype, def_zone)
+	SIGNAL_HANDLER
+	if(damagetype != respectivedamage)
+		return
+	var/mob/living/carbon/human/H = owner
+	var/suitarmor = H.getarmor(null, respectivedamage) / 100
+	var/suit = owner.get_item_by_slot(ITEM_SLOT_OCLOTHING)
+	damagetaken = amount * (1 - suitarmor)
+	if(damagetaken <= 0)
+		return
+	if(!suit)
+		damagetaken = amount
+	if(damagetaken <= shieldhealth)
+//		owner.visible_message("<span class='warning'>The shield deflects [damagetaken] out of [amount] of [damagetype] using their armor of [inherentarmor] base and [suitarmor] from [suit]!</span>")
+		shieldhealth = shieldhealth - damagetaken
+		amount = 0
+		var/shielddiagnostics = shieldhealth / 100
+		if(shielddiagnostics < 0.20 && faltering != 1)
+			faltering = 1
+			owner.visible_message("<span class='warning'>The shield around [owner] begins to falter.</span>")
+		return
+	else
+		qdel(src)
+	return
+
+/datum/status_effect/interventionshield/be_replaced()
+	var/mob/living/carbon/human/L = owner
+	switch(respectivedamage)
+		if(RED_DAMAGE)
+			L.physiology.red_mod /= 0.0001
```

Comment:
As I've said long time ago, we should probably find a way to make it work normally(early return before damage is actually taken by mob) without this mess.

Reply:
How would i hypothetically do this. Cancel a damage taken by mob proc?